machine-learning/Dockerfile

```diff
@@ -1,18 +1,26 @@
-FROM python:3.10
+FROM python:3.10 as builder
 
-ENV TRANSFORMERS_CACHE=/cache
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    PIP_NO_CACHE_DIR=true
 
-WORKDIR /usr/src/app
+COPY requirements.txt ./
+
+RUN python -m venv /opt/venv && \
+    /opt/venv/bin/pip install --upgrade pip setuptools wheel && \
+    /opt/venv/bin/pip install --no-deps -r requirements.txt
+
+FROM python:3.10-slim
 
-RUN python -m venv /opt/venv
-ENV PATH="/opt/venv/bin:$PATH"
+COPY --from=builder /opt/venv /opt/venv
 
-RUN pip install --no-cache-dir torch==1.13.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
-RUN pip install transformers tqdm numpy scikit-learn scipy nltk sentencepiece flask Pillow
-RUN pip install --no-deps sentence-transformers
+ENV TRANSFORMERS_CACHE=/cache \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    PATH="/opt/venv/bin:$PATH"
+
+WORKDIR /usr/src/app
 
 COPY . .
 
-CMD ["python", "src/main.py"]
+CMD ["gunicorn", "src.main:server"]
```

Comment:
Can you update the dev docker compose to use python still? Otherwise we don't have live reload during development.

Reply:
Looks like it is already there in the compose file:
https://github.com/immich-app/immich/blob/main/docker/docker-compose.dev.yml#L33

Reply:
Oh perfect! I didn't think we used that there.